src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -143,12 +144,13 @@ void AddImports(IReadOnlyList<IProjectImportSnapshot> value)
 
         private void OnSourceItemChanged(IProjectSubscriptionUpdate e)
         {
-            foreach (var itemType in e.ProjectChanges)
+            foreach (var itemType in e.ProjectChanges.Where(changes => changes.Value.Difference.AnyChanges))
```

Comment:
Do we also need to do this for other items? Imagine a project that says `<Reference Include="*.dll" /> and a new dll gets dropped into that folder.